pkg/cloud/client.go

```diff
@@ -159,7 +159,8 @@ func NewClientFromConf(conf Config, clientConfig *corev1.ConfigMap) (Client, err
 	}
 
 	clientCacheKey := generateClientCacheKey(conf)
-	if client, exists := clientCache.Get(clientCacheKey); exists {
+	if item := clientCache.Get(clientCacheKey); item != nil {
+		var client interface{} = item.Value()
 		return client.(Client), nil
```

Comment:
Casting was required with previous version of ttlcache. We should be able to directly return `item.Value()` here.
```suggestion
		return item.Value(), nil
```